std/assembly/util/string.ts

```diff
@@ -551,21 +551,21 @@ export function isCaseIgnorable(c: u32): bool {
 
 // @ts-ignore: decorator
 @inline
-export function isFinalSigma(buffer: usize, index: i32, len: i32): bool {
+export function isFinalSigma(buffer: usize, index: isize, len: isize): bool {
   const lookaheadLimit = 30; // max lookahead limit
   var found = false;
   var pos = index;
   var minPos = max(0, pos - lookaheadLimit);
   while (pos > minPos) {
-    let c = codePointBefore(buffer, pos);
+    let c = codePointBefore(buffer, <i32>pos);
     if (!isCaseIgnorable(c)) {
       if (isCased(c)) {
         found = true;
       } else {
         return false;
       }
     }
-    pos -= i32(c >= 0x10000) + 1;
+    pos -= isize(c >= 0x10000) + 1;
```

Comment:
Was wondering if this can be refactored in a better way to avoid converting between types twice in a loop (in 64 bit)?